pkg/printers/internalversion/printers.go

```diff
@@ -962,27 +963,33 @@ func printPod(pod *api.Pod, options printers.GenerateOptions) ([]metav1.TableRow
 					lastRestartDate = terminatedDate
 				}
 			}
-			if container.State.Waiting != nil && container.State.Waiting.Reason != "" {
+			switch {
+			case container.State.Waiting != nil && len(container.State.Waiting.Reason) > 0:
 				reason = container.State.Waiting.Reason
-			} else if container.State.Terminated != nil && container.State.Terminated.Reason != "" {
-				reason = container.State.Terminated.Reason
-			} else if container.State.Terminated != nil && container.State.Terminated.Reason == "" {
-				if container.State.Terminated.Signal != 0 {
+			case container.State.Terminated != nil:
+				if len(container.State.Terminated.Reason) > 0 {
+					reason = container.State.Terminated.Reason
+				} else if container.State.Terminated.Signal != 0 {
 					reason = fmt.Sprintf("Signal:%d", container.State.Terminated.Signal)
 				} else {
 					reason = fmt.Sprintf("ExitCode:%d", container.State.Terminated.ExitCode)
 				}
-			} else if container.Ready && container.State.Running != nil {
+				if container.State.Terminated.Signal != 0 || container.State.Terminated.ExitCode != 0 {
```

Comment:
I would limit the scope of this PR to the evaluation of the `ExitCode` since the seems to be the root cause of the original issue. Besides, there are situations where the container might expect and handle a specific signal which doesn't mean there was an error (not sure how this is handled at CLI level, would need to check).

Anyway, better fix the issue in this PR and maybe open a discussion in another one if there's the need.

Reply:
IIUC the suggestion it is just to change the check with `if container.State.Terminated.ExitCode != 0 {`. This sounds reasonable, as signals aren't taken into account when determining the Pod's phase (see [getPhase](https://github.com/kubernetes/kubernetes/blob/cd30cf186798faf4088638eeb485fda790f78546/pkg/kubelet/kubelet_pods.go#L1563)). So, in case Signal != "" and ExitCode = 0, I think it is preferred not to report an error.